testkit/external/util.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package external
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/pingcap/tidb/domain"
+	"github.com/pingcap/tidb/parser/model"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/table/tables"
+	"github.com/pingcap/tidb/testkit"
+	"github.com/stretchr/testify/require"
+)
+
+// GetTableByName gets table by name for test.
+func GetTableByName(t *testing.T, tk *testkit.TestKit, db, table string) table.Table {
```

Comment:
Could you please describe how moving out these methods resolve cycle import?

IMO it's quite a bit clumsy now to write `GetTableByName(t, tk, ...)` instead of `tk.GetTableByName(...)`.

Also, this PR is blank to describe the background and motivation to make a non-trivial change. It confuses.

Reply:
We have a feature branch and add some test cases.  but we find some import cycles when adding test cases for example ```domain```, ``` table```, and so on. so we think we should add less dependency in the ```testkit``` pkg. and put more dependency into other package.